### PR TITLE
Improve parsing error messages for strategies.yaml.

### PR DESCRIPTION
**Pull request by ywkaras**
This PR primarily flags errors where an unrecognized key (perhaps due to a misspelling) appears in a YAML.

Error reporting could be further improved by including line numbers with errors whenever possible.
Unfortunately, the yamlcpp library seems to have a bug where it doesn't count comment lines when giving
a line number.

### CODE REVIEW

File: tests/gold_tests/next_hop/strategies_ch2/strategies_ch2.test.py [hunk 1]
```diff
@@ -73,7 +73,7 @@
     'proxy.config.http.cache.http': 0,
     'proxy.config.http.uncacheable_requests_bypass_parent': 0,
     'proxy.config.http.no_dns_just_forward_to_parent': 1,
-    'proxy.config.http.parent_proxy.mark_down_hostdb': 1,
+    'proxy.config.http.parent_proxy.mark_down_hostdb': 0,
```
Comment (ywkaras) [inline tests/gold_tests/next_hop/strategies_ch2/strategies_ch2.test.py:76]: mark_down_hostdb should always be 0 when using strategies.yaml.  This change is not related to the purpose of this PR, just threw it in to avoid a 1-line PR.

**Comment by ywkaras** [comment]
`double` is the only failing Au test.

**Comment by ywkaras** [comment]
[approve ci autest]

**Comment by bneradt** [comment]
[approve ci autest]

File: mgmt/YamlCfg.h [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  //
+  class Map
+  {
+  public:
+    // A YAML::ParserException will be thrown if 'map' isn't actually a map.
+    //
+    explicit Map(YAML::Node map);
+
+    // Get the node for a key.  Throw a YAML::Exception if 'key' is not in the map.  The node for each key in the
+    // map must be gotten at least once.  The lifetime of he char array referenced by passed key must be as long
```
Comment (jrushford) [inline mgmt/YamlCfg.h:46]: Spelling, "The lifetime of **the**"

**Review by jrushford — APPROVED** [review]
Looks good @ywkaras.  passes all unit/regression tests.  I tested by introducing misspelled entries and they are found and the strategy is not loaded nor is the remap.

**Review by jrushford — APPROVED** [review]

**Comment by ywkaras** [comment]
[approve ci ubuntu]
